APL 1/Ej-2/ej-2.sh

```diff
@@ -42,3 +42,95 @@
 # Se implementan funciones                                                              Deseable
 
 ################ MAIN ####################
+
+helpFunction() {
+echo "Para la ejecución correcta del programa, debe ingresar los siguientes parametros"
+echo ""
+echo "--path o -p 'directorio' - Para indicar en que directorio desea que los nombres de las fotografias sean editados."
+echo ""
+echo "--help o -h - Para ver este instructivo."
+echo ""
+echo "Ejemplo de ejecucion: ./ej2.sh -p primer carpeta"
+}
+
+
+validatePath(){
+
+    actualFileName="$(basename -- $photo)"
+    ext="${photo#*.}"
+    dir="$(dirname -- $photo)"
+
+    fileName=$actualFileName
+
+	if [ "${fileName:0:8}" -eq "${fileName:0:8}" ] 2>/dev/null; then
+		if [ "${fileName:8:1}" = "_" ]; then
+			if [ "${fileName:9:6}" -eq "${fileName:9:6}" ] 2>/dev/null; then
+				if [ "${fileName:15:4}" = ".jpg" ]; then
+					isValidFile=1
+				fi
+			fi
+		fi
+	fi
+}
+
+changeName(){
+
+    photos=$(find "$directory" -type f -iname "*.jpg")
+    saveIFS=$IFS
+    IFS=$'\n'
+
+    for photo in $photos
+    do
+        isValidFile=0
+        validatePath
+        if [ $isValidFile -eq 1 ];then
+            year=$(echo ${actualFileName:0:4})
+			month=$(echo ${actualFileName:4:2})
+			day=$(echo ${actualFileName:6:2})
+			newDate="$day-$month-$year"
+
+			dateValue=$year-$month-$day
+			dayName=$(date -d $dateValue +%A)
+
+			hour=$(echo ${actualFileName:9:2})
+
+			if [ $hour -lt 19 ] 
+			then
+				food="almuerzo"
+			else
+				food="cena"
+			fi
+		
+
+			mv $photo $dir/$newDate\ $food\ del\ $dayName.$ext
+        fi
+    done
+
+}
+
+################# PARAMETROS
+
+
+if [ $# -eq 2 -a \( "$1" = "--path" -o "$1" = "-p" \) ];then #VALIDA SI EL USUARIO INGRESA EL PARAMETRO --path O -p
+    if [ -d "$2" ];then
```

Comment:
```suggestion
    if [ -d "$2" -a -r "$2" ];then
```
Agregaría la validación de permiso de lectura, porque por más que sea un directorio va a tirar error si es una carpeta a la cual el user no tiene acceso